### PR TITLE
Cis4

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Client for interacting CIS4 registry contract.

## Changes

- Add a CIS4 client to interact with a CIS4 credential registry contract.
- Add a web3id module to retrieve data from a CIS4 contract in order to validate web3id credentials.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: src/cis4.rs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+}
+
+/// Transaction metadata for CIS-4 update transactions.
+pub type Cis4TransactionMetadata = ContractTransactionMetadata;
+
+#[doc(hidden)]
+#[derive(Debug, Clone, Copy)]
+pub enum Cis4Type {}
+
+pub type Cis4Contract = ContractClient<Cis4Type>;
```
Comment (limemloh) [inline src/cis4.rs:73]: This could use some documentation

File: src/cis4.rs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    #[error("Rejected by the node: {0:?}.")]
+    NodeRejected(crate::types::RejectReason),
+}
+
+/// Transaction metadata for CIS-4 update transactions.
+pub type Cis4TransactionMetadata = ContractTransactionMetadata;
+
+#[doc(hidden)]
+#[derive(Debug, Clone, Copy)]
+pub enum Cis4Type {}
```
Comment (limemloh) [inline src/cis4.rs:71]: This should probably be visible in the documentation and explained.

**Review by limemloh — APPROVED** [review]
Don't forget updating the changelog

File: src/cis4.rs (hunk 1 above)
Comment (abizjak) [reply src/cis4.rs:73]: added

File: src/cis4.rs (hunk 2 above)
Comment (abizjak) [reply src/cis4.rs:71]: added

**Pull request merged**